dask/array/creation.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import absolute_import
+from itertools import count
+
+import numpy as np
+
+from .core import Array
+
+linspace_names = ('linspace-%d' % i for i in count(1))
+arange_names = ('arange-%d' % i for i in count(1))
+
+
+def _get_blocksizes(num, blocksize):
+    # compute blockdims
+    remainder = (num % blocksize,)
+    if remainder == (0,):
+        remainder = tuple()
+    blocksizes = ((blocksize,) * int(num // blocksize)) + remainder
+    return blocksizes
+
+
+def linspace(start, stop, blocksize, num=50):
```

Comment:
At the moment we err if the user doesn't provide a blockshape. @shoyer had
thoughts about changing this default

On Wed, Apr 1, 2015 at 8:24 AM, Blake Griffith <notifications@github.com>
wrote:

> In dask/array/creation.py
> <https://github.com/ContinuumIO/dask/pull/115#discussion_r27579641>:
>
> > +from .core import Array
> > +
> > +linspace_names = ('linspace-%d' % i for i in count(1))
> > +arange_names = ('arange-%d' % i for i in count(1))
> > +
> > +
> > +def _get_blocksizes(num, blocksize):
> > +    # compute blockdims
> > +    remainder = (num % blocksize,)
> > +    if remainder == (0,):
> > +        remainder = tuple()
> > +    blocksizes = ((blocksize,) * int(num // blocksize)) + remainder
> > +    return blocksizes
> > +
> > +
> > +def linspace(start, stop, blocksize, num=50):
>
> Is there a way to choose a default blocksize on the fly? Or do we just use
> some constant default?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/115/files#r27579641>.
>